internal/manifest/manifest.go

```diff
@@ -65,33 +70,33 @@ func (p Policy) Hash() HexString {
 
 // Validate checks the validity of all fields in the reference values.
 func (r ReferenceValues) Validate() error {
-	if r.AKS != nil {
-		if err := r.AKS.Validate(); err != nil {
+	if r.SNP != nil {
+		if err := r.SNP.Validate(); err != nil {
 			return fmt.Errorf("validating AKS reference values: %w", err)
 		}
 	}
-	if r.BareMetalTDX != nil {
-		if err := r.BareMetalTDX.Validate(); err != nil {
+	if r.TDX != nil {
+		if err := r.TDX.Validate(); err != nil {
 			return fmt.Errorf("validating bare metal TDX reference values: %w", err)
 		}
 	}
 
-	if r.BareMetalTDX == nil && r.AKS == nil {
+	if r.TDX == nil && r.SNP == nil {
```

Comment:
Do we want to allow reference values instance to contain both TDX and SNP values or do we only want to allow one or the other?

Reply:
> Do we want to allow reference values instance to contain both TDX and SNP values or do we only want to allow one or the other?

I can't think of anything that speaks against this. But for this PR, only SNP support is implemented anyways.

>  have we considered using two arrays of types SNPReferenceValues and TDXReferenceValues respectively instead of using a single array of generic ReferenceValues which may contain one or two of the former?

This would be a trade-off between complexity in the manifest as a whole, or within each entry in `ReferenceValues`. I personally favor the latter here, but I'm open to hear more. I'm also not really sure how that would make it impossible to have an empty entry without validation.

Reply:
> > Do we want to allow reference values instance to contain both TDX and SNP values or do we only want to allow one or the other?
> 
> I can't think of anything that speaks against this. But for this PR, only SNP support is implemented anyways.

This PR validates the TDX reference values and also allows reference values that don't contain SNP reference values. I'm aware that we don't support attestation for TDX yet, right now I only care about the reference values structure itself.

> 
> > have we considered using two arrays of types SNPReferenceValues and TDXReferenceValues respectively instead of using a single array of generic ReferenceValues which may contain one or two of the former?
> 
> This would be a trade-off between complexity in the manifest as a whole, or within each entry in `ReferenceValues`. I personally favor the latter here, but I'm open to hear more.

I'd prefer the former. With your proposed changes the manifest looks like this:
```json
...
  "ReferenceValues": [
    {
      "snp": {
        "MinimumTCB": {
          "BootloaderVersion": 3,
          "TEEVersion": 0,
          "SNPVersion": 8,
          "MicrocodeVersion": 115
        },
        "TrustedMeasurement": "0e5b276722d7625d1e6ceaf48371a4537c64e2c3c149663ff6be87a37c9bd3951c64050af8da87c73a88e9dac9a21e41"
      }
    }
  ],
...
```

If a user wants to use multiple reference values for SNP, they'd have to add multiple reference values entries:
```json
...
  "ReferenceValues": [
    {
      "snp": {
        "MinimumTCB": {
          "BootloaderVersion": 3,
          "TEEVersion": 0,
          "SNPVersion": 8,
          "MicrocodeVersion": 115
        },
        "TrustedMeasurement": "0e5b276722d7625d1e6ceaf48371a4537c64e2c3c149663ff6be87a37c9bd3951c64050af8da87c73a88e9dac9a21e41"
      }
    },
    {
      "snp": {
        "MinimumTCB": {
          "BootloaderVersion": 4,
          "TEEVersion": 0,
          "SNPVersion": 8,
          "MicrocodeVersion": 123
        },
        "TrustedMeasurement": "41c5f8dadbab06887c133a8457bf6897462879b62cb4552189aed78a4b92625b551225f8019e4a403f6e5e3f92af3516"
      }
    }
  ],
...
```

If they also want to use TDX, they can either add a TDX entry to an existing reference values entry:
```json
...
  "ReferenceValues": [
    {
      "snp": {
        "MinimumTCB": {
          "BootloaderVersion": 3,
          "TEEVersion": 0,
          "SNPVersion": 8,
          "MicrocodeVersion": 115
        },
        "TrustedMeasurement": "0e5b276722d7625d1e6ceaf48371a4537c64e2c3c149663ff6be87a37c9bd3951c64050af8da87c73a88e9dac9a21e41"
      },
      "tdx": {
        "TrustedMeasurement": "8dd9080b2a02387ea32e7e2fe0cd6580f02d8a04cbc3b20df01024e5178604ce"
      }
    }
  ],
...
```
or add a new reference values entry:
```json
...
  "ReferenceValues": [
    {
      "snp": {
        "MinimumTCB": {
          "BootloaderVersion": 3,
          "TEEVersion": 0,
          "SNPVersion": 8,
          "MicrocodeVersion": 115
        },
        "TrustedMeasurement": "0e5b276722d7625d1e6ceaf48371a4537c64e2c3c149663ff6be87a37c9bd3951c64050af8da87c73a88e9dac9a21e41"
      },
    },
    {
      "tdx": {
        "TrustedMeasurement": "8dd9080b2a02387ea32e7e2fe0cd6580f02d8a04cbc3b20df01024e5178604ce"
      }
    }
  ],
...
```

My first comment was about how I don't like that there are two different ways to achieve the same thing here (either add TDX to an existing entry with SNP reference values or add a new entry), when I'm not aware of a good reason to allow this. IMHO we should reject the first.

>  I'm also not really sure how that would make it impossible to have an empty entry without validation.

With the current changes, the JSON structure allows the following file:
```json
...
  "ReferenceValues": [
    {
      "snp": {
        "MinimumTCB": {
          "BootloaderVersion": 3,
          "TEEVersion": 0,
          "SNPVersion": 8,
          "MicrocodeVersion": 115
        },
        "TrustedMeasurement": "0e5b276722d7625d1e6ceaf48371a4537c64e2c3c149663ff6be87a37c9bd3951c64050af8da87c73a88e9dac9a21e41"
      },
    },
    {
      // This object is empty
    }
  ],
...
```
We currently need to validate that empty reference values structures don't exist. This is the validation I want to get rid of. 

My second comment proposed a change to this structure to solve this:
```json
...
  "ReferenceValues":  {
    "snp": [{
        "MinimumTCB": {
          "BootloaderVersion": 3,
          "TEEVersion": 0,
          "SNPVersion": 8,
          "MicrocodeVersion": 115
        },
        "TrustedMeasurement": "0e5b276722d7625d1e6ceaf48371a4537c64e2c3c149663ff6be87a37c9bd3951c64050af8da87c73a88e9dac9a21e41"
    },
    {
        "MinimumTCB": {
          "BootloaderVersion": 4,
          "TEEVersion": 0,
          "SNPVersion": 8,
          "MicrocodeVersion": 123
        },
        "TrustedMeasurement": "41c5f8dadbab06887c133a8457bf6897462879b62cb4552189aed78a4b92625b551225f8019e4a403f6e5e3f92af3516"
    }],
    "tdx": [{
      "TrustedMeasurement": "8dd9080b2a02387ea32e7e2fe0cd6580f02d8a04cbc3b20df01024e5178604ce"
    }]
  },
...
```

Not only does this solve the problem of potentially having both SNP and TDX values in the same `ReferenceValues` instance, it also makes it impossible to have an empty `ReferenceValues` because `ReferenceValues` doesn't exist anymore. Of course we still need to validate the values in the `tdx` and `snp` keys, but that's something we'd have to with both strategies.

Reply:
I'd argue that my proposal has lower complexity because we don't need to look at every single reference values entry to figure out which validator(s) to use, we could just look at the key and pass all entries to either the SNP or TDX creation functions. Also, the user doesn't need to repeat the `"snp"`/`"tdx"` keys for every entry.